Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2297,22 +2300,46 @@ void CGContextShowGlyphsWithAdvances(CGContextRef context, const CGGlyph* glyphs
 #pragma endregion
 
 #pragma region Drawing Operations - Basic Shapes
+
+HRESULT __CGContext::ClearRect(CGRect rect) {
+    PushBeginDraw();
+    auto endDraw = wil::ScopeExit([this]() { PopEndDraw(); });
```

Comment:
prefer `return PopEndDraw()` at the end; otherwise, we lose the HRESULT from EndDraw.